tests/dicts/keypath/test_keypath_dict_list_wildcard.py

```diff
@@ -0,0 +1,502 @@
+        self.assertEqual(b.get("b[*]", 1), 1)
+
+    def test_wildcard_get_with_nested_list(self):
+        d = {
+            "a": {
+                "b": [
+                    {
+                        "c": 1,
+                        "d": [1, 2, 3, [0]],
+                    },
+                    {
+                        "c": 2,
+                        "d": [4, 5, 6, [0]],
+                    },
+                    {
+                        "c": 3,
+                        "d": [7, 8, 9, [0]],
+                    },
+                ],
+            },
+        }
+        b = KeypathDict(d)
+
+        self.assertEqual(b.get("a.b[0].d[*]"), [1, 2, 3, [0]])
+        self.assertEqual(b.get("a.b[1].d[*]"), [4, 5, 6, [0]])
+        self.assertEqual(b.get("a.b[2].d[*]"), [7, 8, 9, [0]])
+
+        self.assertEqual(
+            b["a.b[*]"],
+            [
+                {
+                    "c": 1,
+                    "d": [1, 2, 3, [0]],
+                },
+                {
+                    "c": 2,
+                    "d": [4, 5, 6, [0]],
+                },
+                {
+                    "c": 3,
+                    "d": [7, 8, 9, [0]],
+                },
+            ],
+        )
+        self.assertEqual(b.get("a.b[2].d[3][*]"), [0])
+
+    def test_wildcard_list_indexes_with_quotes(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+        self.assertEqual(b.get("a['*']"), [1, 2, 3])
+        self.assertEqual(b.get('a["*"]'), [1, 2, 3])
+
+    def test_wildcard_pop_with_flat_list(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+
+        self.assertEqual(b.pop("a[*]"), [1, 2, 3])
+        self.assertEqual(b, {})
+
+    def test_wildcard_pop_with_flat_list_and_default(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+
+        with self.assertRaises(KeyError):
+            b.pop("b[*]")
+        self.assertEqual(b.pop("b[*]", 4), 4)
+
+    def test_wildcard_pop_with_wrong_index(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+        with self.assertRaises(KeyError):
+            b.pop("b[*]")
+
+    def test_wildcard_pop_with_wrong_index_and_default(self):
+        d = {
+            "a": [1, 2, 3],
+        }
+        b = KeypathDict(d)
+        self.assertEqual(b.pop("b[*]", 6), 6)
+
+    def test_wildcard_pop_with_nested_list(self):
+        d = {
+            "a": {
+                "b": [
+                    {
+                        "c": 1,
+                        "d": [1, 2, 3, [0]],
+                    },
+                    {
+                        "c": 2,
+                        "d": [4, 5, 6, [0]],
+                    },
+                ],
+            },
+        }
+        b = KeypathDict(d)
+
+        self.assertEqual(b.pop("a.b[0].d[-1][*]"), [0])
+        self.assertEqual(b["a.b[0].d"], [1, 2, 3])
```

Comment:
I think this is wrong, the previous `pop` call removes all elements of the last item of `d`, so the last element of `d` should be an empty list (and the `d` value should be `[1, 2, 3, []]`), do you agree?

Reply:
Oh I see, I implemented it as following ```a.pop("a.b") ==  a.pop("a.b[*]")```. It doesn't make sense to have same functionality under different calls.